libfwupd/fwupd-client-sync.c

```diff
@@ -1997,6 +1997,56 @@ fwupd_client_download_bytes (FwupdClient *self,
 	return g_steal_pointer (&helper->bytes);
 }
 
+/**
+ * fwupd_client_download_file:
+ * @self: A #FwupdClient
+ * @url: the remote URL
+ * @file: a #GFile
+ * @flags: #FwupdClientDownloadFlags, e.g. %FWUPD_CLIENT_DOWNLOAD_FLAG_NONE
+ * @cancellable: the #GCancellable, or %NULL
+ * @error: the #GError, or %NULL
+ *
+ * Downloads data from a remote server. The fwupd_client_set_user_agent() function
+ * should be called before this method is used.
```

Comment:
Why not actually check for this and raise an error if it's not done?

Reply:
g_return_val_if_fail style, or GError?

Reply:
/me added the former, i guess it's a programmer error, not a client error